tests/system/common.py

```diff
@@ -800,7 +801,53 @@ def task_by_name(tasks, name):
         if task['name'] == name:
             return task
 
-    assert False, "Did not find task with name %s in this list of tasks: %s" % (name, tasks,)
+    assert False, "Did not find task with name {} in this list of tasks: {}".format(name, tasks)
+
+
+def assert_app_in_all_domains(app, regions=None, zones=None):
+    """Asserts that all tasks of the given app are distributed in all of the given region and/or zone
+    :param app_id: The app instance, as obtained by `marathon_client.get_app(...)`
+    :param region: The regions to test (can be None, String, List or Tuple)
+    :param zone: The zones to test (can be None, String, List or Tuple)
+    """
+
+    if type(regions) is str:
+        regions = [regions]
+    if type(zones) is str:
+        zones = [zones]
+
+    # Sanitize agent domains and user input
+
+    slave_domains = get_cluster_agent_domains()
+
+    assert len(slave_domains) > 0, "Did not find any agents in the DC/OS cluster"
```

Comment:
Please use the matchers from Precisely:

```python
assert_that(agent_domains, has_len(greater_than(0)))
```

This will ease debugging.

Reply:
How is `"agent_domains must be grater than 0"` is more verbose than `"Did not find any agents in the DC/OS cluster"` 😄